examples/connext_dds/printing_qos/c/printing_publisher.c

```diff
@@ -0,0 +1,336 @@
+    publisher = DDS_DomainParticipant_create_publisher(
+        participant, &DDS_PUBLISHER_QOS_DEFAULT, NULL /* listener */,
+        DDS_STATUS_MASK_NONE);
+    if (publisher == NULL) {
+        fprintf(stderr, "create_publisher error\n");
+        publisher_shutdown(participant, &topic_qos, &publisher_qos, &writer_qos);
+        return -1;
+    }
+
+    /* Test DDS_PublisherQos_to_string_w_params */
+    retcode = DDS_Publisher_get_qos(publisher, &publisher_qos);
+    if (retcode != DDS_RETCODE_OK) {
+        fprintf(stderr, "get publisher qos");
+        publisher_shutdown(participant, &topic_qos, &publisher_qos, &writer_qos);
+        return -1;
+    }
+    retcode = DDS_PublisherQos_to_string_w_params(
+            &publisher_qos,
+            str,
+            &strSize,
+            DDS_PUBLISHER_QOS_PRINT_ALL,
+            &printFormat);
+    if (retcode != DDS_RETCODE_OK) {
+        fprintf(stderr, "publisher qos to string");
+        publisher_shutdown(participant, &topic_qos, &publisher_qos, &writer_qos);
+        return -1;
+    }
+    str = DDS_String_alloc(strSize);
+    if (str == NULL) {
+        fprintf(stderr, "String allocation");
+        publisher_shutdown(participant, &topic_qos, &publisher_qos, &writer_qos);
+        return -1;
+    }
+    retcode = DDS_PublisherQos_to_string_w_params(
+            &publisher_qos,
+            str,
+            &strSize,
+            DDS_PUBLISHER_QOS_PRINT_ALL,
+            &printFormat);
+    if (retcode != DDS_RETCODE_OK) {
+        fprintf(stderr, "publisher qos to string");
+        publisher_shutdown(participant, &topic_qos, &publisher_qos, &writer_qos);
+        DDS_String_free(str);
+        return -1;
+    }
+    printf("%s", str);
+    DDS_String_free(str);
+    str = NULL;
+
+    /* Register type before creating topic */
+    type_name = printingTypeSupport_get_type_name();
+    retcode = printingTypeSupport_register_type(
+        participant, type_name);
+    if (retcode != DDS_RETCODE_OK) {
+        fprintf(stderr, "register_type error %d\n", retcode);
+        publisher_shutdown(participant, &topic_qos, &publisher_qos, &writer_qos);
+        return -1;
+    }
+
+    /* To customize topic QoS, use
+    the configuration file USER_QOS_PROFILES.xml */
+    topic = DDS_DomainParticipant_create_topic(
+        participant, "Example printing",
+        type_name, &DDS_TOPIC_QOS_DEFAULT, NULL /* listener */,
+        DDS_STATUS_MASK_NONE);
+    if (topic == NULL) {
+        fprintf(stderr, "create_topic error\n");
+        publisher_shutdown(participant, &topic_qos, &publisher_qos, &writer_qos);
+        return -1;
+    }
+
+    /* Test DDS_TopicQos_to_string */
+    retcode = DDS_Topic_get_qos(topic, &topic_qos);
+    if (retcode != DDS_RETCODE_OK) {
+        fprintf(stderr, "get topic qos");
+        publisher_shutdown(participant, &topic_qos, &publisher_qos, &writer_qos);
+        return -1;
+    }
+    retcode = DDS_TopicQos_to_string(&topic_qos, str, &strSize);
```

Comment:
```
/*
 * First, we pass NULL for the str argument. This will cause the API to update the strSize argument to contain the required size of the str buffer. We then allocate a buffer of that size and obtain the QoS string. The DDS_TopicQos_to_string API only prints differences with respect to the document default values for the TopicQos object.
 */
```

Reply:
Add a similar comment above to the usage of DDS_PublisherQos_to_string_w_params.
Explain that using the to_string_w_params we have more control over what is printed, otherwise only differences with respect to the document default for PublisherQoS is printed.